.github/workflows/typespec-validation-all.yaml

```diff
@@ -58,7 +63,7 @@ jobs:
       - uses: actions/checkout@v4
         with:
           # Convert placeholder string 'default' to '', which tells the action to checkout the default ref.
-          ref: ${{ matrix.ref != 'default' && matrix.ref || '' }}
+          ref: ${{ matrix.ref == 'next' && 'typespec-next' || '' }}
```

Comment:
What does the `&& 'typespec-next' do here? Isn't it a clever way to return it as a value?

Reply:
This is the recommended way to implement something like a "ternary operator" in GH Expressions, using "and" and "or" expressions.  If the condition is true, it returns the first value (which must be non-falsy), else the second value.

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#example